tests/internal/e2e/run_e2e_tests.sh

```diff
@@ -1,13 +1,76 @@
 #!/bin/bash
 ROOT=$(dirname "$(realpath "$0")")
 
+# Get the directory of the current script
+SCRIPT_DIR=$(dirname "$0")
+
+# Define the path to the specs file relative to the script directory
+SPEC_PATH="$SCRIPT_DIR/test/specs_config.ts"
+
+# Check if the specs file exists
+if [ ! -f "$SPEC_PATH" ]; then
+  echo "The file $SPEC_PATH does not exist."
+  exit 1
+fi
+
+# Compile the specs file 
+npx tsc "$SPEC_PATH"
+
+# Path to the compiled JavaScript file
+JS_FILE="./test/specs_config.js"
+
 activate_24_release () {
   export VECTORCAST_DIR=/vcast/release24
   export PATH=/vcast/release24:$PATH
   export ENABLE_ATG_FEATURE=TRUE
   echo "Vcast 24 is activated"
 }
 
+set_specs_params() {
+  # Check if RUN_GROUP_NAME is set
+  if [ -z "$RUN_GROUP_NAME" ]; then
+    echo "RUN_GROUP_NAME is not set. Please set it and try again."
+    exit 1
+  fi
+
+  # Check if the compiled JavaScript file exists
+  if [ ! -f "$JS_FILE" ]; then
+    echo "Compiled JavaScript file not found! Please compile the TypeScript file first."
+    exit 1
+  fi
+
+  # Extract environment variables for the given group using the compiled JavaScript file
+  env_vars=$(node -e "
```

Comment:
Does `require` have to be a hard-coded string, or could it be an argument?

Basically, I feel like it would be nicer if we just committed this snippet of code vs. calling `node -e`.

Reply:
Adressed in 37b4bb4951ced74173e6f06790b959641bd4d1c2 and deleted hardocded exports in 97e6eb7d71236b35712a55095f21026478d6dff6.